packages/next-urql/src/useQuery.ts

```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import { AnyVariables, createRequest, useQuery as orig_useQuery } from 'urql';
+import { useUrqlValue } from './useUrqlValue';
+
+const transportKeys = ['data', 'error'];
+export const useQuery: typeof orig_useQuery = args => {
+  const request = createRequest(
+    args.query,
+    (args.variables || {}) as AnyVariables
+  );
+  useUrqlValue(request.key);
+
+  const [result, execute] = orig_useQuery(args);
+
+  const transported: any = {};
+  for (const key of transportKeys) {
+    // @ts-ignore
+    transported[key] = result[key];
+  }
+
+  useUrqlValue(request.key, transported);
```

Comment:
this reads a bit wonky, I agree but basically we use `useUrqlValue` the first time to read whether or not there's data to append to the ssrExchange and the second time is mainly used on the server to write data. We could split this out into two hooks, I didn't want to conditionally call mainly as that is agains the rules of hooks